src/core/router/util.js

```diff
@@ -44,6 +44,10 @@ export const isAbsolutePath = cached(path => {
   return /(:|(\/{2}))/g.test(path);
 });
 
+export const removeParams = cached(path => {
+  return path.split(/[?#]/)[0];
```

Comment:
could give more details about why you wanna just keep the base path?

Reply:
Yes, I agree. can your describe your changes.

Reply:
this code fix the bug #1136:



1. open a page like 'https://docsify.js.org/#/configuration'

2. click a sub-sidebar link, or click a heading in the main conten, like 'markdown', the url in browser address changes to: 'https://docsify.js.org/#/configuration?id=markdown'

3. using F5 refresh this page, or open a new tab as address 'https://docsify.js.org/#/configuration?id=markdown' in brower.

4. now, sidebar and main body lost sync scroll.

5. because in the function highlight(path)  in scroll.js, does not handle the `id` in `path`;
so , it can't find the corresponding link in sidebar;
so, sidebar and main body does not sync scroll.

6. so, I using this removeParams function, extract base url of the page from the hole url.
```
input: https://docsify.js.org/#/configuration?id=markdown
or: https://docsify.js.org/#/configuration#markdown

output: https://docsify.js.org/#/configuration
```

7. then, docsify can find the corresponding link from urls with `id` param, so, it can sync scroll always.

Reply:
I didnt test the change locally (lack of time, sorry ! )
can you confirm that with this change, will the `id` in the sidebar be highlighted after refresh?

Reply:
yes. I tested this bugfix at firefox/chrome/edge/ie , and place a temp site at https://kepan.org/bugfix
when you click a sidebar item or main body heading, then refresh the browser, or using a url with `id` param like https://kepan.org/bugfix/#/configuration?id=name , the sidebar item always highlighted, and sync scroll.